supervisor/homeassistant/module.py

```diff
@@ -313,46 +322,114 @@ async def backup(self, secure_tar_file: SecureTarFile) -> None:
                 "Preparing backup of Home Assistant Core failed. Check HA Core logs."
             )
 
-        def _write_tarfile():
-            with secure_tar_file as tar_file:
-                # Backup system
-                origin_dir = Path(self.sys_config.path_supervisor, FOLDER_HOMEASSISTANT)
-
-                # Backup data
-                atomic_contents_add(
-                    tar_file,
-                    origin_dir,
-                    excludes=HOMEASSISTANT_BACKUP_EXCLUDE,
-                    arcname=".",
-                )
+        with TemporaryDirectory(dir=self.sys_config.path_tmp) as temp:
+            temp_path = Path(temp)
 
-        try:
-            _LOGGER.info("Backing up Home Assistant Core config folder")
-            await self.sys_run_in_executor(_write_tarfile)
-            _LOGGER.info("Backup Home Assistant Core config folder done")
-        finally:
+            # Store local configs/state
             try:
-                await self.sys_homeassistant.websocket.async_send_command(
-                    {ATTR_TYPE: WSType.BACKUP_END}
-                )
-            except HomeAssistantWSError:
-                _LOGGER.warning(
-                    "Error during Home Assistant Core backup. Check HA Core logs."
-                )
+                write_json_file(temp_path.joinpath("homeassistant.json"), self._data)
+            except ConfigurationFileError as err:
+                raise HomeAssistantError(
+                    f"Can't save meta for Home Assistant Core: {err!s}", _LOGGER.error
+                ) from err
+
+            # Backup data config folder
+            def _write_tarfile():
+                with tar_file as backup:
+                    # Backup metadata
+                    backup.add(temp, arcname=".")
+
+                    # Backup data
+                    atomic_contents_add(
+                        backup,
+                        self.sys_config.path_homeassistant,
+                        excludes=HOMEASSISTANT_BACKUP_EXCLUDE,
+                        arcname="data",
+                    )
 
-    async def restore(self, secure_tar_file: SecureTarFile) -> None:
+            try:
+                _LOGGER.info("Backing up Home Assistant Core config folder")
+                await self.sys_run_in_executor(_write_tarfile)
+                _LOGGER.info("Backup Home Assistant Core config folder done")
+            finally:
+                try:
+                    await self.sys_homeassistant.websocket.async_send_command(
+                        {ATTR_TYPE: WSType.BACKUP_END}
+                    )
```

Comment:
AFAIK we can not allow this to fail, we have to retry until this completes, since if this fails the DB would not accept more entries.
Isn't that right @agners ?

Reply:
We did not retry before afaik, so I think it is fine as it is now.

I mean, it should not fail anyways, unless the Core went down, in which case we can't really do anything anyways.